ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/PauseResumeIntegrationTest.java

```diff
@@ -0,0 +1,279 @@
+        + "QUERIES;").get(0)).getQueries().get(0).getId().toString();
+    RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "PAUSE "
+        + queryId + ";");
+    assertThatEventually(supplier, equalTo(9));
+
+    // Restart server
+    REST_APP.stop();
+    REST_APP.start();
+
+    // Verify PAUSED state -- eventually
+    assertThatEventually(this::getPausedCount, equalTo(1));
+
+    // Verify number of processed records
+    assertThatEventually(supplier, equalTo(9));
+
+    // Then:
+    String queryId2 = ((Queries) RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "SHOW "
+        + "QUERIES;").get(0)).getQueries().get(0).getId().toString();
+    RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "RESUME " + queryId2 + ";");
+    assertThatEventually(this::getPausedCount, equalTo(0));
+    assertThatEventually(this::getRunningCount, equalTo(1));
+
+    // Adding more after restart.
+    TEST_HARNESS.produceRows(PAGE_VIEW_TOPIC, PAGE_VIEWS_PROVIDER2, KAFKA, JSON,
+        System::currentTimeMillis);
+
+    assertThatEventually(supplier, equalTo(14));
+
+    List <KsqlEntity> showQueries3 = RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "SHOW "
+        + "QUERIES;");
+    assertThat(((Queries) showQueries3.get(0)).getQueries().get(0).getStatusCount().getStatuses()
+        .get(KsqlQueryStatus.RUNNING), equalTo(1));
+  }
+
+  // Parameterize this entire class to run with multiple REST instances?
+  // @Test
+  public void pauseResumeShouldWorkWithMultipleServers() {
+
+  }
+
+  private void createQuery(final String suffix) {
+    RestIntegrationTestUtil.makeKsqlRequest(
+        REST_APP,
+        "CREATE STREAM " + SINK_STREAM + suffix
+            + " WITH (kafka_topic='" + SINK_TOPIC + suffix + "',format='json')"
+            + " AS SELECT * FROM " + PAGE_VIEW_STREAM + ";"
+    );
+    TEST_HARNESS.getKafkaCluster().waitForTopicsToBePresent(SINK_TOPIC + suffix);
+  }
+
+  @NotNull
+  private Supplier<Integer> getSupplier(int streamNumber) {
+    return () -> RestIntegrationTestUtil.makeQueryRequest(REST_APP,
+        "select * from " + SINK_STREAM + streamNumber + ";",Optional.empty()).size();
+  }
+
+  private int getPausedCount() {
+    return getCount(KsqlQueryStatus.PAUSED);
+  }
+
+  private int getRunningCount() {
+    return getCount(KsqlQueryStatus.RUNNING);
+  }
+
+  private int getCount(KsqlQueryStatus status) {
+    try {
+
+    List <KsqlEntity> showQueries2 = RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "SHOW "
+        + "QUERIES;");
+    System.out.println("Show queries " + showQueries2.stream().map( q ->
+            ((Queries)q).getQueries().get(0).getStatusCount().toString())
+        .collect(Collectors.joining("\n")));
+
+    // This failing shows that the command topic isn't being read completely?
+    return ((Queries) showQueries2.get(0)).getQueries().get(0).getStatusCount().getStatuses()
+        .get(status);
+    } catch (Exception e) {
+      return 0;
```

Comment:
nit: return `-1` (or some other nonsensical value) to indicate this error, rather than a plausible value (even though this value similarly fails tests).